smartsim/_core/_install/builder.py

```diff
@@ -309,9 +311,9 @@ def __init__(
         self.rai_install_path: t.Optional[Path] = None
 
         # convert to int for RAI build script
-        self.torch = 1 if build_torch else 0
-        self.tf = 1 if build_tf else 0   # pylint: disable=invalid-name
-        self.onnx = 1 if build_onnx else 0
+        self.torch: _TBoolInt = 1 if build_torch else 0
+        self.tf: _TBoolInt = 1 if build_tf else 0  # pylint: disable=invalid-name
+        self.onnx: _TBoolInt = 1 if build_onnx else 0
```

Comment:
should we mark these attributes as protected now that we have the public `fetch_xxx` properties to avoid unintended access?

Reply:
Made them private! I don't think anyone is using these attrs outside of this class, but just in case, I added some read-only `build_*` properties in case someone ever needs to introspect whether or not a backend is going to be built, but removes the ability for them to modify it making the class just a bit more static.